packages/react-components/src/components/VideoGallery/HorizontalGallery.tsx

```diff
@@ -0,0 +1,208 @@
+      });
+      setMaxTiles(maxTiles);
+    }
+  }, [containerWidth, isSmallScreen, leftGutter, rightGutter]);
+
+  const tileSizeStyle = useMemo(
+    () => ({
+      minHeight: `${tileHeight}px`,
+      minWidth: `${tileWidth}px`,
+      maxHeight: `${tileHeight}px`,
+      maxWidth: `${tileWidth}px`
+    }),
+    [tileWidth, tileHeight]
+  );
+
+  const pageButtonSizeStyle = useMemo(
+    () => ({
+      minWidth: '1.75rem',
+      minHeight: `${tileHeight}px`,
+      maxWidth: '1.75rem',
+      maxHeight: `${tileHeight}px`
+    }),
+    [tileHeight]
+  );
+
+  const maxPages = useMemo(() => {
+    return Math.ceil(participants.length / maxTiles) - 1;
+  }, [maxTiles, participants.length]);
+
+  const changePage = (page: number): void => {
+    if (page < 0) {
+      setPage(0);
+    } else if (page >= maxPages) {
+      setPage(maxPages);
+    } else {
+      setPage(page);
+    }
+  };
+
+  const defaultOnRenderParticipants = useMemo(() => {
+    // If user provided a custom onRender function return that function.
+    if (onRenderRemoteVideoTile) {
+      return participants?.map((participant) => onRenderRemoteVideoTile(participant));
+    }
+
+    const start = page * maxTiles;
+    const end = start + maxTiles;
+    return participants?.slice(start, end).map((participant): JSX.Element => {
+      const remoteVideoStream = participant.videoStream;
+      return (
+        <Stack key={participant.userId} className={mergeStyles(tileSizeStyle)}>
+          <RemoteVideoTile
+            key={participant.userId}
+            userId={participant.userId}
+            onCreateRemoteStreamView={hideRemoteVideoStream ? undefined : onCreateRemoteStreamView}
+            onDisposeRemoteStreamView={hideRemoteVideoStream ? undefined : onDisposeRemoteStreamView}
+            isAvailable={hideRemoteVideoStream ? false : remoteVideoStream?.isAvailable}
+            renderElement={hideRemoteVideoStream ? undefined : remoteVideoStream?.renderElement}
+            remoteVideoViewOption={hideRemoteVideoStream ? undefined : remoteVideoViewOption}
+            isMuted={participant.isMuted}
+            isSpeaking={participant.isSpeaking}
+            displayName={participant.displayName}
+            onRenderAvatar={onRenderAvatar}
+            showMuteIndicator={showMuteIndicator}
+          />
+        </Stack>
+      );
+    });
+  }, [
+    onRenderRemoteVideoTile,
+    page,
+    maxTiles,
+    participants,
+    tileSizeStyle,
+    onCreateRemoteStreamView,
+    onDisposeRemoteStreamView,
+    hideRemoteVideoStream,
+    remoteVideoViewOption,
+    onRenderAvatar,
+    showMuteIndicator
+  ]);
+
+  return (
+    <div ref={containerRef}>
+      <Stack
+        horizontal
+        horizontalAlign="start"
+        tokens={{ childrenGap: '0.5rem' }}
+        className={mergeStyles(horizontalGalleryContainerStyle, { paddingLeft: leftGutter, paddingRight: rightGutter })}
+      >
+        {maxTiles && page > 0 && !isSmallScreen ? (
+          <DefaultButton
```

Comment:
Pull these whole default button out into components defined somewhere else on this file, this big chunk should just be muchly simplified:
{showLeftButton && <LeftButton onClick={() => changePage(page - 1)} />}
{defaultOnRenderParticipants}
{showRightButton && <RightButton onClick={() => changePage(page + 1)} />}